packages/manager/modules/pci/src/projects/project/kubernetes/components/autoscaling/autoscaling.controller.js

```diff
@@ -0,0 +1,140 @@
+    this.ovhSubsidiary = coreConfig.getUser().ovhSubsidiary;
+    this.autoscalingInfoLink = this.getAutoscalingInfoLink();
+  }
+
+  $onInit() {
+    this.setLowestHighestLimit();
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      this.model.nodes.lowest.value,
+      this.model.nodes.desired.value,
+      this.model.nodes.highest.value,
+    );
+  }
+
+  getAutoscalingInfoLink() {
+    return AUTOSCALING_LINK[this.ovhSubsidiary] || AUTOSCALING_LINK.DEFAULT;
+  }
+
+  resetScaleValuesToDefault() {
+    const { lowest, highest } = this.model.nodes;
+    lowest.min = SCALE_DEFAULT_VALUES.LOWEST_MIN_VALUE;
+    lowest.value = SCALE_DEFAULT_VALUES.LOWEST_MIN_VALUE;
+
+    highest.value = SCALE_DEFAULT_VALUES.HIGHEST_VALUE;
+    highest.max = SCALE_DEFAULT_VALUES.HIGHEST_MAX_VALUE;
+  }
+
+  setLowestHighestLimit() {
+    const { lowest, desired, highest } = this.model.nodes;
+    lowest.max = desired.value;
+    highest.min = desired.value;
+  }
+
+  onAutoscaleChanged(autoscale) {
+    if (!autoscale.status) {
+      this.resetScaleValuesToDefault();
+    } else {
+      this.setLowestHighestLimit();
+    }
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onAutoscalePoolChanged,
+      autoscale,
+    );
+  }
+
+  onLowestValueChanged(min) {
+    const { lowest, desired, highest } = this.model.nodes;
+
+    highest.min = min.value;
+
+    if (min.value >= desired.value) {
+      lowest.max = min.value;
+    }
+
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      min.value,
+      desired.value,
+      highest.value,
+    );
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onLowestPoolValueChanged,
+      min,
+    );
+  }
+
+  onDesiredValueChanged(need) {
+    const { lowest, highest } = this.model.nodes;
+
+    lowest.max = need.value;
+    highest.min = need.value;
+
+    if (need.value <= lowest.value) {
+      lowest.value = need.value;
+    }
+
+    if (need.value >= highest.value) {
+      highest.value = need.value;
+    }
+
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      lowest.value,
+      need.value,
+      highest.value,
+    );
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onDesiredPoolValueChanged,
+      need,
+    );
+  }
+
+  onHighestValueChanged(max) {
+    const { lowest, desired, highest } = this.model.nodes;
+
+    lowest.max = max.value;
+
+    if (max.value <= desired.value) {
+      highest.min = max.value;
+    }
+
+    this.model.isValidScale = AutoscalingController.isValidKubeScale(
+      this.model.autoscale,
+      lowest.value,
+      desired.value,
+      max.value,
+    );
+
+    AutoscalingController.triggerHandlerFunction(
+      this.onHighestPoolValueChanged,
+      max,
+    );
+  }
+
+  static triggerHandlerFunction(fn, data) {
+    if (fn instanceof Function) fn(data);
```

Comment:
I think if the function binding is set to `&?` this is not needed

Reply:
the `&?` is efficient if we passed or not a callback function via the component attribute.
In my case i implement the on-change handlers into the component. So if parent passed a callback fn it will override the implemented one into the child component, so just one will be triggered not both.
For this case i implement this function, if parent passed a callback i checked if it's a function and trigger it.

Reply:
@mohammed-zahaf I'm having trouble understanding, doesn't `&?`  binding already catch the case if function is undefined ? Maybe we don't need to check if the binding is a function instead of calling it

Reply:
It do it perfectly, but.
Case to consider:
   Case1: we have this: `onLowestPoolValueChange: '&?',`
   if we dont pass a parameter `data-on-lowest-pool-value-change` the onLowestPoolValueChange will be trigger from 
   autoscaling component.
   Case2: we have this: `onLowestPoolValueChange: '&?',`
    if we passed a parameter `data-on-lowest-pool-value-change` the onLowestPoolValueChange will be triggered only 
    from parent not from autoscaling component.
    
You can try it from your side.